Sources/ClientRuntime/Endpoints/DefaultEndpointResolver.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct SmithyHTTPAPI.Endpoint
+import struct SmithyHTTPAPI.Headers
+
+public struct DefaultEndpointResolver<Params: EndpointsRequestContextProviding> {
```

Comment:
This is a reusable endpoint resolver.  It's the same as the previously code-generated endpoint resolver, except:
- It takes a generic set of params instead of hard-coded.
- It does not include generated code to convert params to a `EndpointsRequestContext`, that has been moved to params itself.